dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceConfig.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// Required configuration for Azure OpenAI chat completion with data.
+/// More information: <see href="https://learn.microsoft.com/azure/cognitive-services/openai/quickstart"/>
+/// </summary>
+public sealed class OpenAIServiceConfig
+{
+    /// <summary>
+    /// Azure OpenAI deployment name, see <see href="https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource"/>
+    /// </summary>
+    public string? DeploymentName { get; set; }
```

Comment:
There was opposite opinion regarding that, about merging Azure and OpenAI classes together:
https://github.com/microsoft/semantic-kernel/issues/3155

![image](https://github.com/microsoft/semantic-kernel/assets/13853051/6b97723b-ef33-4759-8122-3a9361097321)

Reply:
@dmytrostruk I don't agree at this point, as moving this abstraction introduce confusion for folks that are using OpenAI and never used Azure and have to wonder what `EndPoint` or `DeploymentName` mean, with nullable fields that will never be used. Not sure if this accomodation is really necessary.

See what Azure SDK did for the UserMessages, SystemMessages, FunctionMessages and AssistantMessages, they have specific classes and that was an important move that brought clarity, before that I didn't know that those specific Roles had different set of Properties...

Reply:
I implemented just one to keep it simple.

Reply:
@RogerBarreto Just sharing other perspective :) My position was to keep Azure/OpenAI service classes separate in order to differentiate them for things like logging. Ideally, it would be great to have separate `Clients` for Azure and OpenAI for more control and flexibility, but it's not the case at the moment. So, I don't have strong preference for `Config` class in this case.